### ISSUE TITLE
Add status command or switch

### ISSUE THREAD

**Issue by petobens** [abraunegg/onedrive#112]
**Is your feature request related to a problem? Please describe.**

Forgive if there is an easy way to do this but i haven't found it (other than monitoring the logs):  when using the `--monitor` I would like to know if one drive is up to date, syncing or has encountered a problem (mainly to show this in my status bar)

**Describe the solution you'd like**
A status command or switch that returns something like "syncing", "up-to-date" and "failed" (for instance with the dropbox client i can do `dropbox-cli status`)

Thanks in advance!

**Comment by abraunegg** [comment]
Due to how the application works, providing a 'status' when running the application in the background (monitor mode) - this will cause database access issues.

This will require some creative thinking to come up with a way to provide something like this.

**Comment by petobens** [comment]
Fair enough. I'll continue parsing the logfile in the mean time. Thanks!

**Comment by fgauvrit** [comment]
Hi,

here is a little bash script, using "yad" package that could help to monitor an manage easily your onedrive client :

--------
#! /bin/bash

export ANS="0"

while [ "$ANS" != "4" ]; do
	yad --fixed --text-align=fill --buttons-layout=spread --center --on-top --form \
					--title="Click on one button to execute action on OneDrive client" \
					--button="<span color=\"green\"><big><b>(Re)Start</b></big></span>":1 \
					--button="<span color=\"red\"><big><b>Stop</b></big></span>":2 \
					--button="<span color=\"blue\"><big><b>Monitor</b></big></span>":3 \
					--button="<big><b>OK</b></big>":4
	export ANS="$?"

	case "$ANS" in
 		1)
    			/usr/bin/systemctl --user restart onedrive
        		;;

 		2)
			/usr/bin/systemctl --user stop onedrive
       			;;
 		3)
			xterm -fa "Monospace" -fs 11 -e '/usr/bin/journalctl --user-unit onedrive -f' &
			gnome-system-monitor -r &
       			;;
	esac
done

exit 0
--------

I hope i helps.

Regards.

**Comment by abraunegg** [comment]
@petobens
Currently in 'master' (due to be in v2.2.2) is the capability for desktop notifications. Whilst this initially shows errors / problems syncing, its a starting point for desktop notifications.

Will also be looking at a command line switch to indicate if everything is up-to-date as well

**Comment by abraunegg** [comment]
@petobens 
Can you help review PR https://github.com/abraunegg/onedrive/pull/307
```
git clone https://github.com/abraunegg/onedrive.git
cd onedrive
git fetch origin pull/307/head:pr307
git checkout pr307
make
make install
```
This work is not complete as yet. 

What this CLI switch will do is check OneDrive for any pending changes. No checking of local changes as yet.

Example:
**OneDrive changes pending**
```
./onedrive --display-sync-status 
Initializing the Synchronization Engine ...
Local directory is out of sync with OneDrive
```
**OneDrive specific folder sync status**
```
./onedrive --display-sync-status --single-directory 'asdf'
Initializing the Synchronization Engine ...
No pending remote changes - selected directory is in sync
```
**After a full sync**
```
./onedrive --display-sync-status 
Initializing the Synchronization Engine ...
No pending remote changes - in sync
```

Any feedback is greatly appreciated.

**Comment by petobens** [comment]
Ok. I''ll give it a shot in these days. Thanks
